#!/usr/bin/env python
PACKAGE = "lynx_driver"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("rate", double_t, 0, "Update rate HZ",    50.0, 0,   100.0)
gen.add("inverse_control", bool_t, 0, "Inverse control output direction",    True)

gen.add("kp", double_t, 0, "P coefficient",    2.0, 0,   10000.0)
gen.add("ki", double_t, 0, "I coefficient",    0.0, 0,   10000.0)
gen.add("kd", double_t, 0, "D coefficient",    0.1, 0,   10000.0)
gen.add("gain", double_t, 0, "Output gain",    50.0, -10000,   10000.0)

gen.add("process_smooth_factor", double_t, 0, "Use weighted average of measurements to smooth the signal",    0.0, 0.0,   1.0)
gen.add("control_smooth_factor", double_t, 0, "Use weighted average of control to smooth the output",    0.8, 0.0,   1.0)
gen.add("control_max_acceleration", double_t, 0, "Maximum allowed acceleration of control signal (in units per second",  550.0, 0.0,   10000.0)

gen.add("output_limit_min", double_t, 0, "Output minimum limit",    500.0, -100000,   100000.0)
gen.add("output_limit_max", double_t, 0, "Output maximum limit",    2500.0, -100000,   100000.0)

gen.add("input_limit_min", double_t, 0, "Input maximum limit",    -10000.0, -100000,   100000.0)
gen.add("input_limit_max", double_t, 0, "Input maximum limit",    10000.0, -100000,   100000.0)

exit(gen.generate(PACKAGE, "lynx_driver", "Pid"))